ipatests/test_webui/test_host.py

```diff
@@ -407,11 +435,353 @@ def test_indirect_associations(self):
+
+        # emtpy CSR
+        csr_add = 'div[name="certificate"] button[name="add"]'
+        csr_add_btn = self.find(csr_add, By.CSS_SELECTOR, strict=True)
+        csr_add_btn.click()
+        self.wait()
+        self.dialog_button_click('ok')
+        self.assert_field_validation_required()
+        self.dialog_button_click('cancel')
+
+        # invalid CSR
+        csr_add = 'div[name="certificate"] button[name="add"]'
+        csr_add_btn = self.find(csr_add, By.CSS_SELECTOR, strict=True)
+        csr_add_btn.click()
+        self.wait()
+        self.fill_textarea('new_cert', host.csr_invalid)
+        self.dialog_button_click('ok')
+        dialog_info = self.get_dialog_info()
+        self.wait()
+        if host.csr_invalid_error in dialog_info['text']:
+            self.dialog_button_click('cancel')
+            self.dialog_button_click('cancel')
+
+        # other hostname CSR
+        self.action_list_action('request_cert', confirm=False)
+        self.assert_dialog()
+        self.fill_text("textarea[name='csr']", host.csr_other_host)
+        self.dialog_button_click('issue')
+        dialog_info = self.get_dialog_info()
+        if host.csr_other_host_error in dialog_info['text']:
+            self.dialog_button_click('cancel')
+            self.dialog_button_click('cancel')
+
+        # cleanup
+        self.delete(ENTITY, [self.data2])
+
+    @screenshot
+    def test_keytab(self):
+        """ Test keytab """
+        self.init_app()
+        self.add_record(ENTITY, self.data2)
+        # provision keytab
+        kt_tmp = '/tmp/test.keytab'
+        hostname = self.data2['pkey']
+        realm = self.config.get('ipa_realm')
+        principal = 'host/{}@{}'.format(hostname, realm)
+        self.run_cmd_on_ui_host('/usr/sbin/ipa-getkeytab '
+                                '-p {} '
+                                '-k {} '.format(principal, kt_tmp))
+        self.navigate_to_record(hostname, entity=ENTITY)
+        self.wait(3)
+        enroll = 'div[name="has_keytab"] label[name="present"]'
+        self.assert_text(enroll, host.krb_enrolled)
+        # test cancel button there bz818665
+        self.action_list_action('unprovision', confirm=False)
+        self.dialog_button_click('cancel')
+        # unprovision keytab
+        self.action_list_action('unprovision', confirm=False)
+        self.dialog_button_click('unprovision')
+        self.wait_for_request(n=4)
+        self.facet_button_click('refresh')
+        enroll = 'div[name="has_keytab"] label[name="missing"]'
+        self.assert_text(enroll, host.krb_not_enrolled)
+        # cleanup
+        self.delete(ENTITY, [self.data2])
+
+    def test_search(self):
+        self.init_app()
+        actions = ActionChains(self.driver)
+        self.navigate_to_entity(ENTITY)
+        self.add_record(ENTITY, [self.data2, self.data3])
+        # positive search filter
+        self.fill_text('div.search-filter input[name="filter"]',
```

Comment:
It might be better to create ui_driver method called e.g.
```python
def fill_search_filter(self, value, parent=None):
   search_field_s = '.search-filter input[name=filter]'
   if not parent:
        parent = self.get_form()
   self.fill_text(search_field_s, value, parent)
   
Would be useful for other tests.